tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -6657,8 +6660,9 @@ bool HLSLExternalSource::MatchArguments(
       }
 
       // Verify that the final results are in bounds.
-      CAB(uCols > 0 && uCols <= MaxVectorSize && uRows > 0 &&
-              uRows <= MaxVectorSize,
+      CAB((uCols > 0 && uRows > 0 &&
+           ((uCols <= MaxVectorSize && uRows <= MaxVectorSize) ||
+            (SM->IsSM69Plus() && uRows == 1))),
```

Comment:
How is this change relevant to allowing native LLVM vectors?  Shouldn't this be part of #7143 instead?